### PR TITLE
Development refactor/listeners

### PR DESCRIPTION
**Pull request by KDwevedi**

### CODE REVIEW

File: cogs/listeners/member_events_cog.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from interfaces.supabase import SupabaseInterface
+
+class MemberEventsListener(commands.Cog):
+    def __init__(self, bot) -> None:
+        self.bot = bot
+        super().__init__()
+
+    @commands.Cog.listener("on_member_join")
+    async def on_member_join(self, member: discord.Member):
+        SupabaseInterface().updateContributor(member)
```
Comment (tushar5526) [inline cogs/listeners/member_events_cog.py:13]: Interfaces are generally meant to be extended and not make instances of. Maybe you can refactor to making a helper class and define static methods on it.

File: interfaces/supabase.py [hunk 2]
```diff
@@ -29,19 +31,33 @@ def update(self, update, query_key, query_value):
     def insert(self, data):
         data = self.client.table(self.table).insert(data).execute()
```
Comment (tushar5526) [inline interfaces/supabase.py:32]: Possible case of `self.table` being equal to `''`

File: helpers.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import discord
```
Comment (tushar5526) [inline helpers.py:1]: Import not getting used ?

**Comment by tushar5526** [comment]
A bunch of observations:

- Generally, python codes use `camel_case` for variable and function declaration. 
- I see type hints in some places. You can use mypy to catch a few more bugs proactively. 
- I can see a lot of unnecessary imports. You can use a combination of `isort` and `flake8` to fix these.
- Also, it would be better integrate `black` for code formatting. 

These linters can be added in as a `pre-commit` and in `GHAs` 

Here is a PR you can refer to for pre-commit setup - https://github.com/Flagsmith/flagsmith-python-client/pull/70
Also attaching a link for Abstrac Base Classes (ABCs) in Python - https://realpython.com/inheritance-composition-python

**Review by tushar5526 — CHANGES_REQUESTED** [review]
Interfaces are generally meant to be extended and not make instances of. Maybe you can refactor to making a helper class and define static methods on it.

**Comment by KDwevedi** [comment]
Tasks:
- [ ] Refactor Supabase interface into a helper class
- [ ] linters, camelCase
- [ ] fixing imports

File: helpers.py (hunk 3 above)
Comment (KDwevedi) [reply helpers.py:1]: made import more specific

File: interfaces/supabase.py (hunk 2 above)
Comment (KDwevedi) [reply interfaces/supabase.py:32]: addressed in supabase refactor, class no longer specifies table on initialisation

File: cogs/listeners/member_events_cog.py (hunk 1 above)
Comment (KDwevedi) [reply cogs/listeners/member_events_cog.py:13]: refactored into helpers/supabaseClient

**Comment by KDwevedi** [comment]
Hey @tushar5526 made some changes
- set up flake8, black, isort
- cleaned imports
- refactored supabase class as supabaseClient

Could you take another look?

**Comment by tushar5526** [comment]
Looks fine to me, a bit more nitpicking but I see `Supabase().function_call()` being used at multiple places. Basically you are trying to namespace all the Supabase helper functions at one place, therefore defining those methods as static methods is much more cleaner.

Also marking the `Singleton` pattern if you want a global instance of Supabase being used. I am approving the PR - you can do these changes as per the priority of this ticket. Thanks!

**Comment by KDwevedi** [comment]
Appreciate the nitpicking!

**Pull request merged**